src/error.rs

```diff
@@ -69,3 +88,20 @@ impl From<IoError> for Error {
         Error::Io(err)
     }
 }
+
+/// A trait that, if supported, returns incomplete image pixel data
+pub trait TryRecover {
+    /// Returns incomplete image pixel data if supported. Otherwise, simply passes through the original value.
+    fn try_recover(self) -> Self;
+}
+
+impl TryRecover for Result<Vec<u8>> {
+    fn try_recover(self) -> Self {
+        self.or_else(|err| match err {
+            Error::Recoverable { pixels, .. } => {
+                Ok(pixels)
+            },
+            _ => Err(err),
+        })
```

Comment:
I would rather provide this as an inherent method of `Error`:

```
impl Error {
    pub fn partial_recover(self) -> Result<Vec<u8>, Error>;
}
```

Each usage site can easily emulate the above trait by calling `Result::{map_,}or_else` themselves which is even slightly more powerful by allowing other `Ok`-types than `Vec<u8>`.